framework/source/class/qx/event/Utils.js

```diff
@@ -103,26 +103,26 @@ qx.Class.define("qx.event.Utils", {
       tracker.promise = newPromise;
       return tracker.promise;
     },
-    
+
     /**
      * Equivalent of `promise.then()`
-     * 
+     *
      * @param tracker {Object} the tracker object
      * @param fn {Function} the function to call when previous promises are complete
      * @return {qx.Promise?} the new promise, or the return value from `fn` if no promises are in use
      */
     then: qx.core.Environment.select("qx.promise", {
       "true": function(tracker, fn) {
         if (tracker.rejected) {
-          return;
+          return null;
         }
         if (tracker.promise) {
           var self = this;
-          this.__push(tracker, tracker.promise.then(function() {
+          this.__push(tracker, tracker.promise.then(function(result) {
             if (tracker.rejected) {
               return null;
             }
-            var result = fn();
+            var result = fn(result);
```

Comment:
This line is declaring `result`, so is then passing `undefined` (`result`) to `fn`. Something fishy going on here...

Reply:
Maybe it's supposed to pass `tracker.result` ?

Reply:
`result` is passed in so the bug is that it shouldn't be declaring it again with `var`; fixed